TestUtilities/ProtobufComparison/src/main/java/com.coursesketch.test.utilities/ProtobufComparison.java

```diff
@@ -170,9 +333,65 @@ private void compareFields(final Object expectedValue, final Object actualValue,
      * @return A string representing the failed message.
      */
     private String createFailMessage(final Descriptors.FieldDescriptor field, final Object expectedValue, final Object actualValue) {
-        if (expectedValue == null) {
-            return "Expected no value for field <" + field.getFullName() + "> but instead got value <" + actualValue + ">";
+        if (field.isRepeated()) {
+            return createFailListMessage(field, expectedValue, actualValue);
+        } else {
```

Comment:
This might be better:
`} else  (expectedValue == null) {
                 return "Expected no value for field [" + field.getFullName() + "] but instead got value <" + actualValue + ">";
             }
             return "Expected <" + expectedValue + "> but got <" + actualValue + "> for field [" + field.getFullName() + "]";
`